neurolib/models/multimodel/builder/base/network.py

```diff
@@ -0,0 +1,717 @@
+
+        :param mass_indices: indices as to which mass to probe
+        :type mass_indices: list|None
+        :return: coupling from all masses indicated in the mass_indices
+        :rtype: dict
+        """
+        mass_indices = mass_indices or np.arange(len(self.masses)).tolist()
+        all_couplings = {}
+        var_idx = 0
+        for mass_idx, mass in enumerate(self.masses):
+            if mass_idx in mass_indices:
+                all_couplings.update({var_idx + k: v for k, v in mass.coupling_variables.items()})
+            var_idx += mass.num_state_variables
+        return all_couplings
+
+    def _derivatives(self, network_coupling=None):
+        """
+        Gather all derivatives from all masses.
+
+        :param network_coupling: dict of network coupling for this node, if
+            None, default network coupling will be used
+        :type network_coupling: dict|None
+        :return: derivatives of the state vector for this node
+        :rtype: list
+        """
+        if network_coupling is None:
+            network_coupling = self.default_network_coupling
+        node_system_equations = []
+        var_idx = 0
+        for mass in self:
+            # get coupling variables
+            coupling_vars = {
+                self._strip_index(key): value
+                for key, value in self.sync_symbols.items()
+                if self._strip_index(key) in mass.required_couplings
+            }
+            coupling_vars.update(
+                {key: value for key, value in network_coupling.items() if key in mass.required_couplings}
+            )
+            mass.idx_state_var = self.idx_state_var + var_idx
+            node_system_equations += mass._derivatives(coupling_vars)
+            var_idx += mass.num_state_variables
+        assert len(node_system_equations) == self.num_state_variables
+        return node_system_equations
+
+
+class SingleCouplingExcitatoryInhibitoryNode(Node):
+    """
+    Basic node with arbitrary number of excitatory and inhibitory populations,
+    but the coupling is through one variable - usually firing rate of the
+    population. Will compute connectivity within node as per population types.
+    This node definition assumes constant delays, i.e. not dependent on time
+    nor dynamics (state variables).
+    """
+
+    name = "Single coupling excitatory vs inhibitory node"
+    label = "1ExcInhNode"
+
+    sync_variables = [
+        "node_exc_exc",
+        "node_inh_exc",
+        "node_exc_inh",
+        "node_inh_inh",
+    ]
+
+    # default network coupling - only EXC to EXC and defaults to 0.0 - no
+    # network
+    default_network_coupling = {"network_exc_exc": 0.0}
+
+    def __init__(
+        self, neural_masses, local_connectivity, local_delays=None,
+    ):
+        """
+        :param neural_masses: list of neural masses in this node
+        :type neural_masses: list[NeuralMass]
+        :param local_connectivity: connectivity matrix for within node
```

Comment:
@caglorithm just now I realised that I am using the convention that any connectivity (or delay for that matter) matrix is `[from, to]`.. but AFAIK `neurolib` uses `[to, from]` as is "standard" in computational neuroscience. is that correct? if so, I'd need to rework all the coupling code...

Reply:
it seems to be the case.. if I can believe the comments in the code :)
```
# Interareal relative coupling strengths (values between 0 and 1), Cmat(i,j) connnection from jth to ith
```
so indeed, connectivity matrices are `[to, from]`... so I'll rework

Reply:
I really do hope so that the comment is correct, I'm gonna check again!